src/containers/LlamaAI/index.tsx

```diff
@@ -2421,11 +2442,19 @@ const ShareModalContent = ({ shareData }: { shareData?: { isPublic: boolean; sha
 
 const ChatControls = memo(function ChatControls({
 	handleSidebarToggle,
-	handleNewChat
+	handleNewChat,
+	sessionId,
+	messages
 }: {
 	handleSidebarToggle: () => void
 	handleNewChat: () => void
+	sessionId: string | null
+	messages: any[]
 }) {
+	const allCharts = messages
+		.filter((m) => m.role === 'assistant' && m.charts?.length > 0)
+		.flatMap((m) => m.charts.map((c: any) => ({ id: c.id, title: c.title })))
+
```

Comment:
there's no point in wrapping `PDFExportButton` in `memo()` if we are not memoizing this array too, will trigger a re render every time unless this is inside `useMemo` or the query to fetch messages